src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -65312,6 +65312,371 @@ static void Test()
+            AssertDiagnosticOptions_38(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_67()
+        {
+            var source =
+@"
+#nullable restore
+#pragma warning enable nullable
+partial class Program
+{
+    static void Test()
+    {
+        var x = M();
+        x = null;
+    }
+}";
+            AssertDiagnosticOptions_38(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_68()
+        {
+            var source =
+@"
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        F(null);
+    }
+}";
+            AssertDiagnosticOptions_01(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_69()
+        {
+            var source =
+@"
+#nullable disable
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        F(null);
+    }
+}";
+            AssertDiagnosticOptions_01(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_70()
+        {
+            var source =
+@"
+#nullable enable
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        F(null);
+    }
+}";
+            AssertDiagnosticOptions_01(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_71()
+        {
+            var source =
+@"
+#nullable safeonly
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        F(null);
+    }
+}";
+            AssertDiagnosticOptions_01(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_72()
+        {
+            var source =
+@"
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        var x = M();
+        x = null;
+    }
+}";
+            AssertDiagnosticOptions_36(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_73()
+        {
+            var source =
+@"
+#nullable disable
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        var x = M();
+        x = null;
+    }
+}";
+            AssertDiagnosticOptions_36(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_74()
+        {
+            var source =
+@"
+#nullable enable
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        var x = M();
+        x = null;
+    }
+}";
+            AssertDiagnosticOptions_36(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_75()
+        {
+            var source =
+@"
+#nullable safeonly
+#pragma warning restore nullable
+partial class Program
+{
+    static void Test()
+    {
+        var x = M();
+        x = null;
+    }
+}";
+            AssertDiagnosticOptions_36(source);
+        }
+
+        [Fact]
+        public void DiagnosticOptions_76()
+        {
+            var source =
+@"
+#pragma warning safeonly nullable
```

Comment:
It's a language question, but this seems strange. I expected `#pragma warning enable safeonly` for this

Reply:
I see that's documented that way in Mads's spec. I'll shoot him an email, but that should not block the PR.

---
In reply to: [245829621](https://github.com/dotnet/roslyn/pull/32208#discussion_r245829621) [](ancestors = 245829621)

Reply:
> It's a language question, but this seems strange. I expected #pragma warning enable safeonly for this

There is a language spec for this here [https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types-specification.md](https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types-specification.md) and I believe the relevant portion has been reviewed before cristmas. 

---
In reply to: [245829621](https://github.com/dotnet/roslyn/pull/32208#discussion_r245829621) [](ancestors = 245829621)